edgelet/iothubservice/src/device.rs

```diff
@@ -81,7 +82,7 @@ where
                 .client
                 .request::<Module, Module>(
                     Method::PUT,
-                    &format!("/devices/{}/modules/{}", &self.device_id, module_id),
+                    &format!("/devices/{}/modules/{}", url_encode(&self.device_id), url_encode(&module_id)),
```

Comment:
We use the `percent_encoding` crate in other areas of the code.

For example: https://github.com/Azure/iotedge/blob/master/edgelet/edgelet-iothub/src/lib.rs#L154

This seems to have a nicer API than the `url::formencoded` support.